### PR TITLE
eth/tracers: add ReturnData in the structLogger tracer's response

### PR DESCRIPTION
**Pull request by jsvisa**
This PR fixes a bug, where the tracer was used through RPC return data of calls were being reported even if user configured `enableReturnData: true`. Return data will be formatted as a hex string.

### CODE REVIEW

File: eth/tracers/logger/logger.go [hunk 1]
```diff
@@ -443,6 +444,13 @@ func formatLogs(logs []StructLog) []StructLogRes {
 			}
 			formatted[index].Stack = &stack
 		}
+		if trace.ReturnData != nil {
```
Comment (s1na) [inline eth/tracers/logger/logger.go:447]: I realize you're breaking it into 32-byte words to be consistent with memory output. I personally don't see a strong reason for it. And memory is usually much bigger. So my vote is just to return a single hex string.

**Review by s1na — COMMENTED** [review]
Good catch seems we're not actually returning returnData even if enabled through the RPC.

File: eth/tracers/logger/logger.go (hunk 1 above)
Comment (s1na) [reply eth/tracers/logger/logger.go:447]: @holiman is there any benefits to chunking it for the fuzzers/evmlab etc?

**Comment by holiman** [comment]
Fuzzers/evmlab doesn't use the logger, for two reasons:

1. It doesn't stream output, but collects all in-memory first, so is likely to crash on large traces.
2. The output is not machine-friendly, like the 32-byte chunking shows. 

So imo it's mostly legacy and unused, and I don't have any opinion about to chunk or not.

**Review by s1na — APPROVED** [review]
Prefixed with 0x. LGTM

**Pull request merged**